File: packages/python/staffkm-sdk/staffkm/resources/applications.py
"""Applications resource."""
from __future__ import annotations

import httpx


class ApplicationsResource:
    def __init__(self, http: httpx.Client):
        self._http = http

    def list(self) -> list[dict]:
        r = self._http.get("/api/v1/applications")
        r.raise_for_status()
        return r.json().get("data") or []

    def create(self, name: str, type: str = "chat", **kwargs) -> dict:
        body = {"name": name, "type": type, **kwargs}
        r = self._http.post("/api/v1/applications", json=body)
        r.raise_for_status()
        return r.json().get("data") or r.json()

    def get(self, application_id: str) -> dict:
        r = self._http.get(f"/api/v1/applications/{application_id}")
        r.raise_for_status()
        return r.json().get("data") or r.json()

    def run(self, application_id: str, inputs: dict) -> dict:
        r = self._http.post(
            f"/api/v1/applications/{application_id}/run",
            json={"inputs": inputs},
        )
        r.raise_for_status()
        return r.json()

File: packages/ts/staffkm-sdk/src/resources/applications.ts
import { AxiosInstance } from 'axios'

export class ApplicationsResource {
  constructor(private http: AxiosInstance) {}

  async list(): Promise<any[]> {
    const r = await this.http.get('/api/v1/applications')
    return r.data?.data ?? []
  }

  async create(name: string, type = 'chat', extra: Record<string, any> = {}): Promise<any> {
    const r = await this.http.post('/api/v1/applications', { name, type, ...extra })
    return r.data?.data ?? r.data
  }

  async get(id: string): Promise<any> {
    const r = await this.http.get(`/api/v1/applications/${id}`)
    return r.data?.data ?? r.data
  }

  async run(id: string, inputs: Record<string, any>): Promise<any> {
    const r = await this.http.post(`/api/v1/applications/${id}/run`, { inputs })
    return r.data
  }
}
